core/local/analysis.js

```diff
@@ -38,37 +38,55 @@ const panic = (context, description) => {
   throw new Error(description)
 }
 
-function analyseEvents (events /*: LocalEvent[] */, pendingChanges /*: LocalChange[] */) /*: LocalChange[] */ {
-  const stopMeasure = measureTime('LocalWatcher#analyseEvents')
-  // OPTIMIZE: new Array(events.length)
-  const changes /*: LocalChange[] */ = []
-  const changesByInode /*: Map<number, LocalChange> */ = new Map()
-  const changesByPath /*: Map<string, LocalChange> */ = new Map()
-  const getChangeByInode = (e) => {
-    const ino = getInode(e)
-    if (ino) return changesByInode.get(ino)
+class LocalChangeMap {
+  /*::
+  changes: LocalChange[]
+  changesByInode: Map<number, LocalChange>
+  changesByPath: Map<string, LocalChange>
+  */
+
+  constructor () {
+    this._clear()
+  }
+
+  _clear () {
+    this.changes = []
+    this.changesByInode = new Map()
+    this.changesByPath = new Map()
+  }
+
+  findByInode (ino /*: ?number */) /*: ?LocalChange */ {
+    if (ino) return this.changesByInode.get(ino)
```

Comment:
Agreed, but it was already named `ino` in the previous version, I didn't change it, so I would say it's out of the scope of this PR.
Also it is already called `ino` in the whole codebase and in `fs.Stats`.
In case we really want to rename it, I'd prefer renaming everything at once.